src/device.rs

```diff
@@ -18,4 +24,34 @@ impl Device {
             ))
         }
     }
+
+    pub fn get_history<P: IsA<Window>>(
+        &self,
+        window: &P,
+        start: u32,
+        stop: u32,
+    ) -> Option<Vec<TimeCoord>> {
+        unsafe {
+            let mut events = ptr::null_mut();
+            let mut n_events = mem::MaybeUninit::uninit();
+            let ret: bool = from_glib(gdk_sys::gdk_device_get_history(
+                self.to_glib_none().0,
+                window.as_ref().to_glib_none().0,
+                start,
+                stop,
+                &mut events,
+                n_events.as_mut_ptr(),
+            ));
+            if ret {
```

Comment:
This is inverted. `if !ret { return None; }` would be correct

Reply:
Good catch!